src/efftester.ml

```diff
@@ -587,21 +660,25 @@ let varRules env s eff size = (* vars have no immediate effect, so 'eff' param i
                (x:s), env |- m : t
     -------------------------------------------- (LAM)
        env |- (fun (x:s) -> m) : s -> t
- *)
-let rec lamRules env u eff size = (* lams have no immediate effect, so 'eff' param is ignored *)
-  let gen s eff t = Gen.(varGen >>= fun x ->
-			 listPermuteTermGenOuter (addVar x s env) t eff (size/2) >>= function
-			   | None   -> return None
-			   | Some m ->
-			     let myeff = imm_eff m in
-			     return (Some (Lambda (Fun (s,myeff,imm_type m),x,s,m)))) in
+*)
+let rec lamRules env u eff size =
+  (* lams have no immediate effect, so 'eff' param is ignored *)
+  let gen s eff t =
+    Gen.(
+      varGen
+      >>= fun x ->
```

Comment:
I would prefer to use `--no-break-infix-before-func` which was designed specifically to get `varGen >>= fun x ->` here, which is the more readable style once you understand the correspondence with `monadic-let x = varGen in`.